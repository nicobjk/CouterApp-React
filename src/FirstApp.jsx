import PropTypes from 'prop-types';
// //const getResult = () => {
//     return ('Hola Taradas!!!')
// };
// Si no hay ninguna dependencia, colocar las funciones fuera del funtional component. 

//const newMessage = [1,2,3,4,5];

export const FirstApp = ( {title , subtitle , name} ) => {
    return (
        <>
            {/*para mostrar una funcion:
            <h1>{ getResult() }</h1> */}
            
            {/*para mostrar datos de un objeto es necesario el JSON.stringify.
            <code>{ JSON.stringify( newMessage ) }</code>*/} 

            <h1 data-testid="test-title"> { title } </h1>
            <p>{ subtitle }</p>
            <p>{ subtitle }</p>
            <p>{ name }</p>  
        </>
    )
}

FirstApp.propTypes = {
    subtitle: PropTypes.string,
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Nicolás Bojko',
    subtitle: 'No hay subtítulo!',
    //title: 'No hay título!',
}