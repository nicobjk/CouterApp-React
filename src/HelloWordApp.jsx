// function App() {
//     // document.createElement...
//     return <h1>Â¡Hola mundo!</h1>
// }

// export default App; 


//El profe prefiere hacerlo de esta manera, no hay un beneficio directo. 
import React from 'react'

export const HelloWordApp = () => {
    return (
        <h1>Hello Word App!</h1>
    )
}
