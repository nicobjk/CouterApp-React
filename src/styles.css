html, body {
    background-color: #21232A ;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.1rem;
    padding: 30px;
    text-align: center;

};

button {
    padding: 15px;
    font-size: 1.5rem;
    border-radius: 35%;
    margin-right: 5px;
}