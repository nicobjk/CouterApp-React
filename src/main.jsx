import React from 'react';
import ReactDOM from 'react-dom/client';
//import { HelloWordApp } from './HelloWordApp';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';
import './styles.css';


ReactDOM.createRoot ( document.getElementById( 'root' )).render(
    <React.StrictMode>
        < CounterApp value = { 0 } />
        {/*<FirstApp title="Hola Taradas!"/>*/}
    </React.StrictMode>
);