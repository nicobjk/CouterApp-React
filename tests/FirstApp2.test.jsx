import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas de <FirstApp/>', () => {

    const title = 'Hola Taradas!!!';
    const subtitle= 'Como estan???'

    test('Debe hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el mensaje "Hola Taradas!!!"', () => {
        render(<FirstApp title={title}/>);
        //screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debe mostrar el titulo en un h1', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('Debe mostrar el sutitulo enviado por props', () => {
        render(
        <FirstApp 
            title={title}
            subtitle={subtitle}   
        />);
        expect(screen.getAllByText(subtitle).length).toBe(2);
    });
});