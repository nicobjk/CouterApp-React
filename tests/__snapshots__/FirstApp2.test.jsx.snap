// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <FirstApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola Taradas!!!
     
  </h1>
  <p>
    No hay subtítulo!
  </p>
  <p>
    No hay subtítulo!
  </p>
  <p>
    Nicolás Bojko
  </p>
</div>
`;
